sycl/include/CL/sycl/ordered_queue.hpp

```diff
@@ -88,15 +88,53 @@ class __SYCL_DEPRECATED__ ordered_queue {
   template <info::queue param>
   typename info::param_traits<info::queue, param>::return_type get_info() const;
 
-  template <typename T> event submit(T cgf) { return submit_impl(cgf); }
+  /// @param Loc is the code location of the submit call (default argument)
+  template <typename T> event submit(T cgf
+  #ifdef SYCL_INSTRUMENTATION_METADATA
+                                     , const cl::sycl::detail::code_location &CodeLoc = cl::sycl::detail::code_location::current()
+  #endif
+                                    ) {
+  #ifndef SYCL_INSTRUMENTATION_METADATA
+    const cl::sycl::detail::code_location &CodeLoc = {};
+  #endif
+    return submit_impl(cgf, CodeLoc);
+  }
 
-  template <typename T> event submit(T cgf, ordered_queue &secondaryQueue) {
-    return submit_impl(cgf, secondaryQueue);
+  template <typename T> event submit(T cgf,
+                                     ordered_queue &secondaryQueue
+  #ifdef SYCL_INSTRUMENTATION_METADATA
+                                     , const cl::sycl::detail::code_location &CodeLoc = cl::sycl::detail::code_location::current()
+  #endif
+                                    ) {
+  #ifndef SYCL_INSTRUMENTATION_METADATA
+    const cl::sycl::detail::code_location &CodeLoc = {};
+  #endif
+    return submit_impl(cgf, secondaryQueue, CodeLoc);
   }
 
-  void wait();
+  /// @param CodeLoc is the code location of the submit call (default argument)
+  void wait(
+  #ifdef SYCL_INSTRUMENTATION_METADATA
+            const cl::sycl::detail::code_location &CodeLoc = cl::sycl::detail::code_location::current()
+  #endif
+  ) {
+  #ifndef SYCL_INSTRUMENTATION_METADATA
+    const cl::sycl::detail::code_location &CodeLoc = {};
+  #endif
+    impl->wait(CodeLoc);
```

Comment:
Please, move this to cpp file.

Reply:
It uses the SYCL_INSTRUMENTATION_METADATA guard, which would make it problematic as this  maybe controlled by the compiler to include or not include the default argument in the public interface. Moving it to the cpp file will have an implementation that is based on the SYCL_INSTRUMENTATION_METADATA macro define at the SYCL runtime build time.

Reply:
@alexbatashev  Ok, in order to move this to the cpp file, I had to introduce a proxy method that captures the code location using these guards and the proxy method is implemented in the cpp file. This should satisfy the requirement. Also, to address @romanovvlad  regarding the SYCL_INSTRUMENTATION_METADATA, the use is reversed and if someone needs the code locations disabled, they have compile with -DDISABLE_SYCL_INSTRUMENTATION_METADATA and locations are on by default.

Reply:
@alexbatashev  Is it OK now?

Reply:
@romanovvlad I still see `impl->wait(CodeLoc)` in browser. @tovinkere is GitHub having hard times showing big PRs or you haven't pushed your changes yet?

Reply:
@alexbatashev  The changes you are looking have been pushed a while ago.  It's odd that you don't see it.